### PR TITLE
feat: Migrate to `objc2`

### PR DESCRIPTION
**Pull request by madsmtm**
I've migrated a lot of the crates Tauri depends on, see backrefererences to https://github.com/tauri-apps/wry/issues/1239,  let's do it for Tauri as well.

The benefit to this part of Tauri's codebase is honestly not that big, I'm doing this mostly for consistency, though it does unblock https://github.com/tauri-apps/tauri/issues/9969.

I haven't signed the commit, it's kind of a hazzle that I don't really want to deal with, feel free to rebase it into oblivion, I don't need the attribution.

### CODE REVIEW

**Comment by madsmtm** [comment]
Not sure I did the changelog entries correctly

File: crates/tauri-runtime-wry/src/webview.rs [hunk 1]
```diff
@@ -13,25 +13,17 @@ mod imp {
...
-
-  pub struct Webview {
-    pub webview: id,
-    pub manager: id,
-    pub view_controller: id,
+    pub webview: *mut c_void,
+    pub manager: *mut c_void,
+    #[cfg(target_os = "macos")]
+    pub ns_window: *mut c_void,
+    #[cfg(not(target_os = "macos"))]
```
Comment (lucasfernog) [inline crates/tauri-runtime-wry/src/webview.rs]:
```suggestion
    #[cfg(target_os = "ios")]
```
just for consistency with the usage

Comment (lucasfernog) [reply crates/tauri-runtime-wry/src/webview.rs]: also makes it easier for us to locate it when we want to support visionOS

**Review by lucasfernog — APPROVED** [review]
this is awesome!

**Pull request merged**